src/core/runner.ts

```diff
@@ -410,6 +407,15 @@ export default class Runner {
       journey.callback({ params: options.params } as any);
       journey.monitor.update(this.monitor?.config);
       journey.monitor.validate();
+
+      /**
+       * Add the tags created through `monitor.use` to the journey
+       */
```

Comment:
Why not add this as part of update monitor which handles all config from 'monitor.use'?